src/learn/perceptron.py

```diff
@@ -10,14 +10,25 @@
                     format='%(asctime)s %(levelname)s: %(message)s',
                     datefmt='%m/%d/%Y %I:%M:%S %p')
 
-class PerceptronLearner():
+class Learner():
 
     def __init__(self, w_vector=None, max_iter=1):
+        """
+        :param w_vector: A global weight vector instance that stores
+         the weight value (float)
+        :param max_iter: Maximum iterations for training the weight vector
+         Could be overridden by parameter max_iter in the method
+        :return: None
+        """
         logging.debug("Initialize PerceptronLearner ... ")
         self.w_vector = w_vector
         self.max_iter = max_iter
+
         return
 
+    def learner_name(self):
```

Comment:
Class attributes are just attributes that all instances of a class will have. They're very commonly used, especially for stuff like this where you know what the value of it is going to be ahead of time. They get a lot more interesting when you start working with descriptors (e.g. through the `property()` function).

Methods are sort of the same way: `learner.learner_name()` is sort of\* the same thing as `Learner.learner_name(learner)`

\* they start differing when you assign something else to `learner.learner_name`, e.g `learner.learner_name = 5`